### PR TITLE
Add module check result tracking to RPC API

### PR DESCRIPTION
**Pull request by acammack-r7**
This adds a few sets and a hash to the `Msf::Simple::Framework` that help keep track of running checks and their eventual results.

Verification
========

- [x] Start the RPC service
- [x] Request `module.check` with `exploit/windows/smb/ms08_067_netapi`
- [x] With the UUID you receive back, poll `module.results` until the status is no longer `ready` or `running`
- [x] With the same UUID, request `module.ack` and verify the results no longer show in `module.results`
- [x] Ditto the above, but with `module.run` and `auxiliary/scanner/rdp/cve_2019_0708_bluekeep`

### CODE REVIEW

File: lib/msf/base/simple/auxiliary.rb [hunk 1]
```diff
@@ -132,12 +160,23 @@ def check_simple(opts)
...
       mod.framework.events.on_module_run(mod)
-      mod.run
+      begin
+        mod.framework.running << run_uuid
+        mod.framework.ready.delete run_uuid
+        result = block.call(mod)
+        mod.framework.results[run_uuid] = {result: result}
+      rescue Exception => e
+        mod.framework.results[run_uuid] = {error: e.to_s}
+        raise
```
Comment (jmartin-r7) [inline lib/msf/base/simple/auxiliary.rb]: Should this raise a specific type or pass `e`?

Comment (acammack-r7) [reply lib/msf/base/simple/auxiliary.rb]: `$!` is the last exception rescued and is what `raise` raises when no arguments are passed inside a `rescue` clause.

**Comment by busterb** [comment]
Is there anything left to do here to promote from draft status?

**Comment by acammack-r7** [comment]
@busterb, testing and verification steps, mostly. One of the internal teams is running into some issues with it, so I definitely need to poke at it a bit more.

File: lib/msf/base/simple/auxiliary.rb [hunk 2]
```diff
@@ -68,21 +68,25 @@ def self.run_simple(omod, opts = {}, &block)
       mod.init_ui(nil, nil)
     end
 
-    ctx = [ mod ]
+    run_uuid = Rex::Text.rand_text_alphanumeric(24)
```
Comment (dwelch-r7) [inline lib/msf/base/simple/auxiliary.rb]: could we rename run_uuid or make it store a real UUID instead? 
I could see this causing confusion if someone (me) was trying to debug wondering why the uuid isn't a uuid :sweat_smile:

Comment (acammack-r7) [reply lib/msf/base/simple/auxiliary.rb]: I largely agree, however Metasploit has a long history of calling things "UUID" that don't meet any of the UUID specs (influenced by a number of things, I imagine, including the historical [`Object#id`](https://ruby-doc.org/core-1.8.7/Object.html#method-i-id) complicating calling anything just `id` and RFC 4122 passing in the year this codebase started). This value is taking the place of `module.uuid` (only 8 alnum bytes!) in the RPC API https://github.com/rapid7/metasploit-framework/blob/master/lib/msf/core/rpc/v10/rpc_module.rb#L671. Then there's [payload UUIDs](https://github.com/rapid7/metasploit-framework/blob/master/lib/msf/core/payload/uuid.rb). Fixing the situation would be nice, but will take some thought and is outside the scope of this PR.

**Comment by busterb** [comment]
## Release Notes

This enhances the module run RPC API to include vulnerability check results in the output.

**Comment by jmartin-r7** [comment]
This enhances #10735 marking `msf5` for now.

**Pull request merged**